server/util.go

```diff
@@ -386,13 +378,11 @@ func (d *resultEncoder) encodeData(src []byte) []byte {
 }
 
 func (d *resultEncoder) encodeWith(src []byte, enc charset.Encoding) []byte {
-	var err error
-	d.buffer, err = enc.Transform(d.buffer, src, charset.OpEncode)
+	data, err := enc.Transform(d.buffer, src, charset.OpEncode)
```

Comment:
Does it mean always copy here?

Reply:
noop, if we can return `src` directly, the `buffer` is not used in `transform`

Reply:
I init the buffer in the new commit, PTAL

Reply:
So if user changed the value of `data`, valued of `src` will follow the change. Seems the behavior of `Transform` still inconsistent.

Reply:
@Defined2014 @xiongjiwei Maybe it is better to add a comment like 'the result is read-only' to `Transform()`.